File: python/Problem125.py
# Problem: 125. Valid Palindrome
# URL: https://leetcode.com/problems/valid-palindrome/
# Difficulty: Easy
# Tags: String


class Solution:
  def isPalindrome(self, s: str) -> bool:
    i: int = 0
    j: int = len(s) - 1
    while i < j:
      while i < j and not s[i].isalnum():
        i += 1
      while i < j and not s[j].isalnum():
        j -= 1
      if s[i].lower() != s[j].lower():
        return False
      i += 1
      j -= 1
    return True

File: cpp/Problem125.cpp
/*
 * Problem: 125. Valid Palindrome
 * URL: https://leetcode.com/problems/valid-palindrome/
 * Difficulty: Easy
 * Tags: String
 */

#include <string>

using namespace std;

class Solution {
public:
	bool isPalindrome(string s) {
		int i = 0;
		int j = s.length() - 1;
		while (i < j) {
			while (i < j && !isalnum(s[i])) {
				i++;
			}
			while (i < j && !isalnum(s[j])) {
				j--;
			}
			if (tolower(s[i]) != tolower(s[j])) {
				return false;
			}
			i++;
			j--;
		}
		return true;
	}
};